distribution/test/itests/test-itests-ddf/src/test/java/ddf/test/itests/catalog/TestFederation.java

```diff
@@ -1424,54 +1447,244 @@ public void testProductDownloadListEmptyWhenNoDownloads() {
-                .match(Condition.get("/services/csw"),
-                        Condition.parameter("request", "GetRecordById"),
-                        Condition.parameter("id", metacardId))
-                .then(getCswRetrievalHeaders(filename),
-                        chunkedContentWithHeaders(resourceData, Duration.ofMillis(500), 0));
-
-        String startDownloadUrl =
-                RESOURCE_DOWNLOAD_ENDPOINT_ROOT.getUrl() + "?source=" + CSW_STUB_SOURCE_ID
-                        + "&metacard=" + metacardId;
+        String startDownloadUrl = setUpDownloadResponse(metacardId, filename, resourceData, NO_RETRIES, delayBetweenRetries);
 
         // @formatter:off
         String downloadId = when().get(startDownloadUrl).then().log().all()
                 .extract().jsonPath().getString("downloadId");
         // @formatter:on
+        assertThat(downloadId, not(isEmptyString()));
 
         String getAllDownloadsUrl = RESOURCE_DOWNLOAD_ENDPOINT_ROOT.getUrl();
 
         // @formatter:off
-        List<Map<String, Object>> downloads = expect("List of active downloads is not empty").within(10, SECONDS)
+        List<Map<String, Object>> downloads = expect("List of active downloads is not empty").within(30, SECONDS)
                 .until(()-> when().get(getAllDownloadsUrl)
                         .then().log().all().extract().body().jsonPath().getList(""), hasSize(1)).lastResult();
         // @formatter:on
 
         Map download = downloads.get(0);
         assertThat(download.size(), is(6));
 
-        assertThat(download.get("downloadId"), is(downloadId));
-        assertThat(download.get("fileName"), is(filename));
-        assertThat(download.get("status"), is("IN_PROGRESS"));
-        int bytesDownloaded = (int) download.get("bytesDownloaded");
-        assertThat(bytesDownloaded, is(greaterThan(0)));
-        assertThat(bytesDownloaded, is(lessThan(resourceData.length() + 1)));
-        assertTrue(((String) download.get("percentDownloaded")).matches("UNKNOWN|[0-9]|[0-9]{2}|100"));
-        assertThat((List<String>) download.get("users"), contains("Guest@Guest@127.0.0.1"));
+        checkDownloadInProgress(download, downloadId, filename, resourceData);
+
+    }
+
+    /**
+     * Similar to the test for one active download but checks that two downloads can be active at once.
+     *
+     * @throws Exception
+     */
+    @Test
+    public void testProductDownloadListWithTwoActiveDownloads() throws IOException {
+
+        String filename1 = "product1.txt";
+        String metacardId1 = generateUniqueMetacardId();
+        String resourceData1 = getResourceData(metacardId1);
+
+        String filename2 = "product2.txt";
+        String metacardId2 = generateUniqueMetacardId();
+        String resourceData2 = getResourceData(metacardId2);
+
+        int delayBetweenRetries = 200;
+
+        String startDownloadUrl1 = setUpDownloadResponse(metacardId1, filename1, resourceData1, NO_RETRIES, delayBetweenRetries);
```

Comment:
It would be neat if we could create a class from this `setUpDownloadResponse` and take in the filename, metacard ID, and resource data in the constructor, set things up and then just call `startDownload()` (which would do what L1503-L1506 currently do).